src/lib/types/model.ts

```diff
@@ -365,3 +365,66 @@ export interface IFeatureStrategySegment {
     featureStrategyId: string;
     segmentId: number;
 }
+
+export interface ISuggestChangeset {
+    id: number;
+    state: string;
+    project: string;
+    environment: string;
+    createdBy?: Pick<User, 'username' | 'imageUrl'>;
+    createdAt?: Date;
+    changes?: ISuggestChange[];
+    events?: ISuggestChangeEvent[];
+    // approvalFlow: IChangeSetApprovalFlow;
+}
+export interface IApprovalFlow {}
+
+export interface IChangeSetApprovalFlow {
+    id: string;
+    flow: IApprovalFlow;
+}
+
+export interface ISuggestChange {
+    id: number;
+    action: SuggestChangeAction;
+    feature: string;
+    payload?: unknown;
+    createdBy?: Pick<User, 'username' | 'imageUrl'>;
+    createdAt?: Date;
```

Comment:
Does it make sense to have a change without a payload? Maybe there's cases where it's not necessary (like enabling/disabling a toggle)? And as with the change set: why are createdBy and createdAt optional?

Reply:
see below